master/internal/resourcemanagers/kubernetes/pods.go

```diff
@@ -614,11 +641,59 @@ func (p *pods) summarize(ctx *actor.Context) map[string]model.AgentSummary {
 			ID:             node.Name,
 			RegisteredTime: node.ObjectMeta.CreationTimestamp.Time,
 			Slots:          slotsSummary,
-			NumContainers:  len(podByNode[node.Name]),
+			NumContainers:  len(podByNode[node.Name]) + len(nodeToTasks[node.Name]),
 			ResourcePool:   "",
 			Addresses:      addrs,
 		}
 	}
 
 	return summary
 }
+
+func (p *pods) getNonDetPods() []k8sV1.Pod {
+	var nonDetPods []k8sV1.Pod
+	pList, err := p.clientSet.CoreV1().Pods("default").List(context.TODO(), metaV1.ListOptions{})
+	if err != nil {
+		return nonDetPods
+	}
+	for _, p := range pList.Items {
+		if _, ok := p.Labels["determined"]; !ok {
+			if p.Spec.NodeName != "" {
+				nonDetPods = append(nonDetPods, p)
+			}
+		}
+	}
+	return nonDetPods
+}
+
+func (p *pods) getNonDetSlots(deviceType device.Type) (map[string][]string, map[string]int) {
+	nodeToTasks := make(map[string][]string)
+	taskSlots := make(map[string]int)
+
+	nonDetPods := p.getNonDetPods()
+	if len(nonDetPods) == 0 {
+		return nodeToTasks, taskSlots
+	}
+	for _, node := range p.currentNodes {
+		nodeToTasks[node.Name] = []string{}
+	}
+
+	for _, p := range nonDetPods {
+		if _, ok := nodeToTasks[p.Spec.NodeName]; !ok {
+			continue
+		}
+		reqs := 0
+		for _, c := range p.Spec.Containers {
+			if deviceType == device.CPU {
+				reqs += int(c.Resources.Requests.Cpu().Value())
```

Comment:
this is a number of vCPUs but we have `p.slotResourceRequests.CPU` to define how many real vCPUs are in one determined slot:
https://github.com/determined-ai/determined/blob/master/master/internal/resourcemanagers/kubernetes/pods.go#L566

so perhaps this should similarly do:

```suggestion
				reqs += int64(float32(c.Resources.Requests.Cpu().MilliValue()) / (1000. * p.slotResourceRequests.CPU))
```

maybe even move that math to helper